Elm 0.17.1 doesn't build

## Issue description

I'm on unstable.  When doing nix-env --upgrade, the Elm compile fails, leaving me with the previous version.  The specific error:

```
[75 of 88] Compiling Elm.Docs         ( src/Elm/Docs.hs, dist/build/Elm/Docs.o )

src/Elm/Docs.hs:102:25: error:
    • No instance for (Num Json.Indent) arising from the literal ‘2’
    • In the ‘confIndent’ field of a record
      In the expression:
        Json.Config
          {Json.confIndent = 2, Json.confCompare = Json.keyOrder keys}
      In an equation for ‘config’:
          config
            = Json.Config
                {Json.confIndent = 2, Json.confCompare = Json.keyOrder keys}
            where
                keys = ["tag", ....]
builder for ‘/nix/store/irjap10vl1idnkk0flmxjncj1nlqnagq-elm-compiler-0.17.1.drv’ failed with exit code 1
cannot build derivation ‘/nix/store/v7w2l6vjngs67z9pxmnnd91z368bw309-elm-0.17.1.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/v7w2l6vjngs67z9pxmnnd91z368bw309-elm-0.17.1.drv’ failed
```

### Steps to reproduce

Have elm installed, execute nix-env --upgrade.  Or presumably not have elm installed, be on unstable, and install it.  

## Technical details

* System: (NixOS: `nixos-version`, Ubuntu/Fedora: `lsb_release -a`, ...)

16.09pre87733.dbd856d (Flounder)

* Nix version: (run `nix-env --version`)

nix-env (Nix) 1.11.2

* Nixpkgs version: (run `nix-instantiate --eval '<nixpkgs>' -A lib.nixpkgsVersion`)

"16.09pre87733.dbd856d"

--- a/pkgs/development/compilers/elm/packages/elm-compiler.nix
+++ b/pkgs/development/compilers/elm/packages/elm-compiler.nix
@@ -1,4 +1,4 @@
-{ mkDerivation, aeson, aeson-pretty, ansi-terminal, ansi-wl-pprint
+{ mkDerivation, aeson, aeson-pretty_0_7_2, ansi-terminal, ansi-wl-pprint
 , base, binary, bytestring, containers, directory, edit-distance
 , fetchgit, filemanip, filepath, HUnit, indents
 , language-ecmascript, language-glsl, mtl, parsec, pretty, process
@@ -16,7 +16,7 @@ mkDerivation {
   isLibrary = true;
   isExecutable = true;
   libraryHaskellDepends = [
-    aeson aeson-pretty ansi-terminal ansi-wl-pprint base binary
+    aeson aeson-pretty_0_7_2 ansi-terminal ansi-wl-pprint base binary
     bytestring containers directory edit-distance filepath indents
     language-ecmascript language-glsl mtl parsec pretty process text
     union-find
@@ -25,7 +25,7 @@ mkDerivation {
     aeson base binary directory filepath process text
   ];
   testHaskellDepends = [
-    aeson aeson-pretty ansi-terminal ansi-wl-pprint base binary
+    aeson aeson-pretty_0_7_2 ansi-terminal ansi-wl-pprint base binary
     bytestring containers directory edit-distance filemanip filepath
     HUnit indents language-ecmascript language-glsl mtl parsec pretty
     process QuickCheck test-framework test-framework-hunit

--- a/pkgs/development/compilers/elm/packages/elm-package.nix
+++ b/pkgs/development/compilers/elm/packages/elm-package.nix
@@ -1,4 +1,4 @@
-{ mkDerivation, aeson, aeson-pretty, ansi-wl-pprint, base, binary
+{ mkDerivation, aeson, aeson-pretty_0_7_2, ansi-wl-pprint, base, binary
 , bytestring, containers, directory, edit-distance, elm-compiler
 , fetchgit, filepath, HTTP, http-client, http-client-tls
 , http-types, mtl, network, optparse-applicative, parallel-io
@@ -16,13 +16,13 @@ mkDerivation {
   isLibrary = true;
   isExecutable = true;
   libraryHaskellDepends = [
-    aeson aeson-pretty ansi-wl-pprint base binary bytestring containers
+    aeson aeson-pretty_0_7_2 ansi-wl-pprint base binary bytestring containers
     directory edit-distance elm-compiler filepath HTTP http-client
     http-client-tls http-types mtl network parallel-io text time
     unordered-containers vector zip-archive
   ];
   executableHaskellDepends = [
-    aeson aeson-pretty ansi-wl-pprint base binary bytestring containers
+    aeson aeson-pretty_0_7_2 ansi-wl-pprint base binary bytestring containers
     directory edit-distance elm-compiler filepath HTTP http-client
     http-client-tls http-types mtl network optparse-applicative
     parallel-io pretty text time unordered-containers vector

